File: src/BreakfastProvider.Api/Models/Requests/InventoryItemRequest.cs
namespace BreakfastProvider.Api.Models.Requests;

public record InventoryItemRequest
{
    public string? Name { get; init; }
    public string? Category { get; init; }
    public decimal Quantity { get; init; }
    public string? Unit { get; init; }
    public decimal ReorderLevel { get; init; }
}

File: java/breakfast-provider-app/src/main/java/io/lemonlion/breakfast/model/request/InventoryItemRequest.java
package io.lemonlion.breakfast.model.request;

import java.math.BigDecimal;

/** Twin of C# {@code InventoryItemRequest} ({@code decimal} -> {@link BigDecimal}; omitted numbers default to 0). */
public record InventoryItemRequest(
        String name, String category, BigDecimal quantity, String unit, BigDecimal reorderLevel) {

    public InventoryItemRequest {
        if (quantity == null) {
            quantity = BigDecimal.ZERO;
        }
        if (reorderLevel == null) {
            reorderLevel = BigDecimal.ZERO;
        }
    }
}
